fips-check.sh

```diff
@@ -9,118 +9,242 @@
 # This should check out all the approved flavors. The command line
 # option selects the flavor. The keep option keeps the output
 # directory.
-#
-# Some variables may be overridden on the command line.
+
+# These variables may be overridden on the command line.
+MAKE="${MAKE:-make}"
+GIT="${GIT:-git -c advice.detachedHead=false}"
+TEST_DIR="${TEST_DIR:-XXX-fips-test}"
+FLAVOR="${FLAVOR:-linux}"
+KEEP="${KEEP:-no}"
+FIPS_REPO="${FIPS_REPO:-git@github.com:wolfssl/fips.git}"
 
 Usage() {
     cat <<usageText
 Usage: $0 [flavor] [keep]
 Flavor is one of:
-    netbsd-selftest
-    marvell-linux-selftest
     linuxv2 (FIPSv2, use for Win10)
-    wolfrand
+    fipsv2-OE-ready (ready FIPSv2)
     solaris
+    netbsd-selftest
+    marvell-linux-selftest
     linuxv5 (current FIPS 140-3)
     fips-ready (ready FIPS 140-3)
     fips-dev (dev FIPS 140-3)
-Keep (default off) retains the XXX-fips-test temp dir for inspection.
+    wolfrand
+Keep (default off) retains the temp dir $TEST_DIR for inspection.
 
 Example:
     $0 windows keep
 usageText
 }
 
-# These variables may be overridden on the command line.
-MAKE="${MAKE:-make}"
-GIT="${GIT:-git -c advice.detachedHead=false}"
-TEST_DIR="${TEST_DIR:-XXX-fips-test}"
-FLAVOR="${FLAVOR:-linux}"
-KEEP="${KEEP:-no}"
-
 while [ "$1" ]; do
   if [ "$1" = 'keep' ]; then KEEP='yes'; else FLAVOR="$1"; fi
   shift
 done
 
-FIPS_REPO='git@github.com:wolfssl/fips.git'
-CRYPT_INC_PATH='wolfssl/wolfcrypt'
-CRYPT_SRC_PATH='wolfcrypt/src'
-
 case "$FLAVOR" in
-linuxv2|fipsv2-OE-ready)
+linuxv2|fipsv2-OE-ready|solaris)
   FIPS_OPTION='v2'
-  FIPS_VERSION='WCv4-stable'
-  CRYPT_VERSION='WCv4-stable'
-  RNG_VERSION='WCv4-rng-stable'
-  WC_MODS=('aes' 'aes_asm' 'cmac' 'des3' 'dh' 'ecc' 'hmac' 'random' 'rsa' 'sha' 'sha256' 'sha3' 'sha512')
-  FIPS_SRCS=('fips.c' 'fips_test.c' 'wolfcrypt_first.c' 'wolfcrypt_last.c')
-  FIPS_INCS=('fips.h')
+  FIPS_FILES=('WCv4-stable'
+    'wolfcrypt/src/fips.c'
+    'wolfcrypt/src/fips_test.c'
+    'wolfcrypt/src/wolfcrypt_first.c'
+    'wolfcrypt/src/wolfcrypt_last.c'
+    'wolfssl/wolfcrypt/fips.h'
+  )
+  WOLFCRYPT_FILES=(
+    'wolfcrypt/src/aes.c:WCv4-stable'
```

Comment:
Note to any other reviewers, while this seems redundant to have the same tag after every file the goal here is to be able to individually version the algo files. If we port for example to OE1 and have to make a changes to aes.c, rsa.c and ecc.c this list might end up looking like:

```
   'wolfcrypt/src/aes.c:OE1-stable' <--------- |
   'wolfcrypt/src/aes_asm.asm:WCv4-stable'
    'wolfcrypt/src/aes_asm.S:WCv4-stable'
    'wolfcrypt/src/cmac.c:WCv4-stable'
    'wolfcrypt/src/des3.c:WCv4-stable'
    'wolfcrypt/src/dh.c:WCv4-stable'
    'wolfcrypt/src/ecc.c:OE1-stable' <--------- |
    'wolfcrypt/src/hmac.c:WCv4-stable'
    'wolfcrypt/src/random.c:WCv4-rng-stable'
    'wolfcrypt/src/rsa.c:OE1-stable' <--------- |
    'wolfcrypt/src/sha.c:WCv4-stable'
    'wolfcrypt/src/sha256.c:WCv4-stable'
    'wolfcrypt/src/sha3.c:WCv4-stable'
    'wolfcrypt/src/sha512.c:WCv4-stable'
    'wolfssl/wolfcrypt/aes.h:WCv4-stable'
    'wolfssl/wolfcrypt/cmac.h:WCv4-stable'
    'wolfssl/wolfcrypt/des3.h:WCv4-stable'
    'wolfssl/wolfcrypt/dh.h:WCv4-stable'
    'wolfssl/wolfcrypt/ecc.h:WCv4-stable'
    'wolfssl/wolfcrypt/hmac.h:WCv4-stable'
    'wolfssl/wolfcrypt/random.h:WCv4-rng-stable'
    'wolfssl/wolfcrypt/rsa.h:WCv4-stable'
    'wolfssl/wolfcrypt/sha.h:WCv4-stable'
    'wolfssl/wolfcrypt/sha256.h:WCv4-stable'
    'wolfssl/wolfcrypt/sha3.h:WCv4-stable'
    'wolfssl/wolfcrypt/sha512.h:WCv4-stable'
    ```
    
    It is NOT a mistake to not be using a variable in the list for the tag.